### PR TITLE
support table metadata loading for H2 dialect

### LINKED ISSUE #9617: Load columns for all tables into metadata
We load rules configured tables into metadata with columns and table name only for un-configured tables now. 
By product design, it is necessary to load all columns for every table.

Load all columns are too slow if using raw JDBC interface. We should create a new SPI for quickly load columns via raw database way, such as load metadata via `information_schema` for MySQL. 

The tasks are: 

- [x] Create SPI for load metadata with native database dialect
- [x] Load all columns for default JDBC loader
- [x] Add MySQL dialect
- [x] Finish index metadata loading for MySQL dialect
- [ ] Add PostgreSQL dialect
- [x] Add SQLServer dialect
- [ ] Add Oracle dialect
- [ ] Add H2 dialect

Some issues are related with imperfect metadata, there are: #9615

I have already finished kernel framework adjust and MySQL implementation.
I just leave other database dialects for volunteer wanted.

Volunteers can reference `MySQLTableMetaDataLoader` and `MySQLTableMetaDataLoaderTest` to implement other dialect of databases.

### PR DESCRIPTION
**Pull request by zhujunxxxxx**
Fixes #9617.

Changes proposed in this pull request:
- table metadata loading for H2 dialect

### CODE REVIEW

**Comment by zhujunxxxxx** [comment]
why? in my local build is success. @terrymanu can you help me check my ci problem.

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoaderTest.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                .executeQuery())
+                .thenReturn(primaryKeys);
+
+        ResultSet generatedInfo = mockGeneratedInfoResultSet();
+        when(dataSource.getConnection().prepareStatement(
+                "SELECT C.TABLE_NAME TABLE_NAME, C.COLUMN_NAME COLUMN_NAME, COALESCE(S.IS_GENERATED, FALSE) IS_GENERATED FROM INFORMATION_SCHEMA.COLUMNS C"
+                        + " RIGHT JOIN INFORMATION_SCHEMA.SEQUENCES S ON C.SEQUENCE_NAME=S.SEQUENCE_NAME WHERE C.TABLE_CATALOG=? AND C.TABLE_SCHEMA=? AND TABLE_NAME NOT IN ('existed_tbl')")
+                .executeQuery())
+                .thenReturn(generatedInfo);
+
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoaderTest.java:96]: unnecessary blank line, please remove.

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoaderTest.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        DataSource dataSource = mockDataSource();
+        ResultSet resultSet = mockTableMetaDataResultSet();
+        when(dataSource.getConnection().prepareStatement(
+                "SELECT TABLE_CATALOG, TABLE_NAME, COLUMN_NAME, DATA_TYPE, TYPE_NAME FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_CATALOG=? AND TABLE_SCHEMA=?").executeQuery()).thenReturn(resultSet);
+        ResultSet indexResultSet = mockIndexMetaDataResultSet();
+        when(dataSource.getConnection().prepareStatement(
+                "SELECT TABLE_CATALOG, TABLE_NAME, INDEX_NAME, COLUMN_NAME FROM INFORMATION_SCHEMA.INDEXES WHERE TABLE_CATALOG=? AND TABLE_SCHEMA=? AND TABLE_NAME IN ('tbl')")
+                .executeQuery())
+                .thenReturn(indexResultSet);
+
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoaderTest.java:58]: Unnecessary blank line, please remove.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Table meta data loader for H2.
+ * @author zhujun
+ */
+public class H2TableMetaDataLoader implements DialectTableMetaDataLoader {
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:43]: It is better to define it as final one.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Table meta data loader for H2.
+ * @author zhujun
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:41]: Sorry, please remove @author.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    private Map<String, Collection<ColumnMetaData>> loadColumnMetaDataMap(final DataSource dataSource, final Collection<String> existedTables) throws SQLException {
+        Map<String, Collection<ColumnMetaData>> result = new HashMap<>();
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(getTableMetaDataSQL(existedTables))) {
+            Map<String, Integer> dataTypes = DataTypeLoader.load(connection.getMetaData());
+
+            Map<String, Collection<String>> tablePrimaryKeys = loadTablePrimaryKeys(dataSource, existedTables);
+            Map<String, Map<String, Boolean>> tableGenerated = loadTableGenerated(dataSource, existedTables);
+
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:85]: There are many needless blank lines, please remove them.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/resources/META-INF/services/org.apache.shardingsphere.infra.metadata.schema.builder.spi.DialectTableMetaDataLoader [hunk 6]
```diff
@@ -16,3 +16,4 @@
 #
 
 org.apache.shardingsphere.infra.metadata.schema.builder.loader.dialect.MySQLTableMetaDataLoader
+org.apache.shardingsphere.infra.metadata.schema.builder.loader.dialect.H2TableMetaDataLoader
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/resources/META-INF/services/org.apache.shardingsphere.infra.metadata.schema.builder.spi.DialectTableMetaDataLoader:19]: A new line below is needed.

**Review by tristaZero — CHANGES_REQUESTED** [review]
Hi @zhujunxxxxx ,
Thanks for your contribution. This PR basically LGTM, please give your attention to my comments to make it better.

**Comment by tristaZero** [comment]
We can get the error info from the gitHub action here,
```
[ERROR] assertDecorateMetaDataContexts(org.apache.shardingsphere.proxy.initializer.impl.GovernanceBootstrapInitializerTest)  Time elapsed: 0.263 s  <<< ERROR!
23701java.lang.NullPointerException
23702	at org.apache.shardingsphere.proxy.initializer.impl.GovernanceBootstrapInitializerTest.assertDecorateMetaDataContexts(GovernanceBootstrapInitializerTest.java:205)
23703
23704[INFO] Running org.apache.shardingsphere.proxy.initializer.impl.StandardBootstrapInitializerTest
23705[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.012 s - in org.apache.shardingsphere.proxy.initializer.impl.StandardBootstrapInitializerTest
23706[INFO] 
23707[INFO] Results:
23708[INFO] 
23709[ERROR] Errors: 
23710[ERROR]   GovernanceBootstrapInitializerTest.assertDecorateMetaDataContexts:205 » NullPointer
23711[INFO] 
23712[ERROR] Tests run: 18, Failures: 0, Errors: 1, Skipped: 0
```

Could you pull the latest code from master branch and give it a try? Look forward to your feedback.

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoaderTest.java (hunk 1 above)
Comment (zhujunxxxxx) [reply shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoaderTest.java:96]: OK, thank you review, I will fix it later.

**Comment by zhujunxxxxx** [comment]
image
@tristaZero there have new problem of CI.

**Comment by tristaZero** [comment]
Hi CI is triggered to rerun.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+        Map<String, Collection<ColumnMetaData>> result = new HashMap<>();
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(getTableMetaDataSQL(existedTables))) {
+            Map<String, Integer> dataTypes = DataTypeLoader.load(connection.getMetaData());
+            Map<String, Collection<String>> tablePrimaryKeys = loadTablePrimaryKeys(dataSource, existedTables);
+            Map<String, Map<String, Boolean>> tableGenerated = loadTableGenerated(dataSource, existedTables);
+            preparedStatement.setString(1, connection.getCatalog());
+            preparedStatement.setString(2, "PUBLIC");
+
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:85]: Please remove it.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(getIndexMetaDataSQL(tableNames))) {
+            preparedStatement.setString(1, connection.getCatalog());
+            preparedStatement.setString(2, "PUBLIC");
+            try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                while (resultSet.next()) {
+                    String indexName = resultSet.getString("INDEX_NAME");
+                    String tableName = resultSet.getString("TABLE_NAME");
+
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:128]: Hi, Same issue here.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+        return existedTables.isEmpty() ? PRIMARY_KEY_META_DATA_SQL
+                : String.format(PRIMARY_KEY_META_DATA_SQL_WITH_EXISTED_TABLES, existedTables.stream().map(each -> String.format("'%s'", each)).collect(Collectors.joining(",")));
+    }
+
+    private Map<String, Collection<String>> loadTablePrimaryKeys(final DataSource dataSource, final Collection<String> tableNames) throws SQLException {
+        Map<String, Collection<String>> result = new HashMap<>();
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(getPrimaryKeyMetaDataSQL(tableNames))) {
+
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:158]: Need to remove this blank line.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+            preparedStatement.setString(1, connection.getCatalog());
+            preparedStatement.setString(2, "PUBLIC");
+
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:161]: Need to remove this blank line.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+                : String.format(GENERATED_INFO_SQL_WITH_EXISTED_TABLES, existedTables.stream().map(each -> String.format("'%s'", each)).collect(Collectors.joining(",")));
+    }
+
+    private Map<String, Map<String, Boolean>> loadTableGenerated(final DataSource dataSource, final Collection<String> tableNames) throws SQLException {
+
+        Map<String, Map<String, Boolean>> result = new HashMap<>();
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(getGeneratedInfoSQL(tableNames))) {
+
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:184]: Need to remove this blank line.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+            preparedStatement.setString(1, connection.getCatalog());
+            preparedStatement.setString(2, "PUBLIC");
+
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:187]: Need to remove this blank line.

**Review by tristaZero — CHANGES_REQUESTED** [review]
Hi, your PR looks complete, nice. :)
Just look at some comment to make it concise.

**Comment by tristaZero** [comment]
@zhujunxxxxx 

Hi, LGTM, it is the right time to merge, nice job.

```
[INFO] Running org.apache.shardingsphere.test.integration.engine.it.dql.GeneralDQLIT
25446[ERROR] Tests run: 1360, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 101.262 s <<< FAILURE! - in org.apache.shardingsphere.test.integration.engine.it.dql.GeneralDQLIT
25447[ERROR] assertExecute[jdbc: dbtbl_with_read_write_splitting -> H2 -> Placeholder -> SELECT * FROM t_order WHERE user_id >= ? AND user_id <= ? AND order_id >= ? AND order_id <= ? ORDER BY user_id, order_id](org.apache.shardingsphere.test.integration.engine.it.dql.GeneralDQLIT)  Time elapsed: 0.02 s  <<< ERROR!
25448java.sql.SQLTransientConnectionException: HikariPool-26 - Connection is not available, request timed out after 30000ms.
25449
25450[INFO] Running org.apache.shardingsphere.test.integration.engine.it.dql.AdditionalDQLIT
25451[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in org.apache.shardingsphere.test.integration.engine.it.dql.AdditionalDQLIT
25452[INFO] Running org.apache.shardingsphere.test.integration.engine.it.dcl.GeneralDCLIT
25453[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.009 s - in org.apache.shardingsphere.test.integration.engine.it.dcl.GeneralDCLIT
25454[INFO] Running org.apache.shardingsphere.test.integration.engine.it.ddl.GeneralDDLIT
25455[INFO] Tests run: 192, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 35.752 s - in org.apache.shardingsphere.test.integration.engine.it.ddl.GeneralDDLIT
25456[INFO] 
25457[INFO] Results:
25458[INFO] 
25459[ERROR] Errors: 
25460[ERROR] org.apache.shardingsphere.test.integration.engine.it.dql.GeneralDQLIT.assertExecute[jdbc: dbtbl_with_read_write_splitting -> H2 -> Placeholder -> SELECT * FROM t_order WHERE user_id >= ? AND user_id <= ? AND order_id >= ? AND order_id <= ? ORDER BY user_id, order_id](org.apache.shardingsphere.test.integration.engine.it.dql.GeneralDQLIT)
25461[ERROR]   Run 1: GeneralDQLIT.assertExecute[jdbc: dbtbl_with_read_write_splitting -> H2 -> Placeholder -> SELECT * FROM t_order WHERE user_id >= ? AND user_id <= ? AND order_id >= ? AND order_id <= ? ORDER BY user_id, order_id] » SQLTransientConnection
25462[INFO]   Run 2: PASS
```
@JiekerTime Please give this a look, looks it is something to do with integrated tests.

**Comment by zhujunxxxxx** [comment]
@tristaZero  integrated tests failed, because datasource getConnection timeout. In class `org.apache.shardingsphere.test.integration.env.datasource.builder.ActualDataSourceBuilder.createHikariCP` create datasource need increase MaximumPoolSize to 10. I test in my local is pass.

**Comment by zhujunxxxxx** [comment]
@tristaZero After analyzing the code, I found that the old datasource connection pool is only 2. when  integrated tests multiple threads acquire a connection concurrently, some threads will wait for a timeout. so just modify MaximumPoolSize  to 10.

**Comment by tristaZero** [comment]
Hi @zhujunxxxxx ,

> After analyzing the code, I found that the old datasource connection pool is only 2. when integrated tests multiple 

But the question is why the MySQL scenario can run well? Please set `it.databases=MySQL` in `env/engine-env.properties` and run `GeneralDQLIT`. As you know, your file will be called in the same position with `MySQLTableMetaDataLoader`, right?

Besides, `loadColumnMetaDataMap` of this PR creates two connections in parallel.

**Comment by zhujunxxxxx** [comment]
> But the question is why the MySQL scenario can run well? 

Yes，I known the reason ,  I debug found `org.apache.shardingsphere.infra.metadata.schema.builder.SchemaBuilder.appendRemainTables`  method run in MySQL scenario  `findDialectTableMetaDataLoader` will return `Optional.empty()` so `appendRemainTables` will go to `appendDefaultRemainTables` metod，it will not use `MySQLTableMetaDataLoader` to load metadata.  because testcase use database type is H2，so in my scenario findDialectTableMetaDataLoader will return `H2TableMetaDataLoader`

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+        return result;
+    }
+
+    private Map<String, Collection<ColumnMetaData>> loadColumnMetaDataMap(final DataSource dataSource, final Collection<String> existedTables) throws SQLException {
+        Map<String, Collection<ColumnMetaData>> result = new HashMap<>();
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(getTableMetaDataSQL(existedTables))) {
+            Map<String, Integer> dataTypes = DataTypeLoader.load(connection.getMetaData());
+            Map<String, Collection<String>> tablePrimaryKeys = loadTablePrimaryKeys(dataSource, existedTables);
```
Comment (JiekerTime) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:81]: How about trying to reuse connections?

**Review by JiekerTime — COMMENTED** [review]
To avoid create more connections, we can try to reuse connections.

**Comment by JiekerTime** [comment]
> @tristaZero integrated tests failed, because datasource getConnection timeout. In class `org.apache.shardingsphere.test.integration.env.datasource.builder.ActualDataSourceBuilder.createHikariCP` create datasource need increase MaximumPoolSize to 10. I test in my local is pass.

IMO, The problem is not caused byMaximumPoolSize . The configuration here is similar to the configuration as follows:
image
I think the reason for the PR is more likely to be caused by other factors such as concurrency.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java (hunk 13 above)
Comment (zhujunxxxxx) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:81]: so, how to reuse connections?

Comment (JiekerTime) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:81]: Take line 81 for an example. If the parameter of loadTablePrimaryKeys()  you pass in is not dataSource but connection, how do you think about it?

**Comment by zhujunxxxxx** [comment]
@tristaZero @JiekerTime I modify `it.databases=MySQL` in `env/engine-env.properties` and run GeneralDQLIT appear same problem.  so, I am sure this it not my problem. It's just that the current scenario was not tested by the previous use case

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java (hunk 13 above)
Comment (zhujunxxxxx) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:81]: Please see my latest reply, this problem appears in MySQL scenario

Comment (zhujunxxxxx) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoader.java:81]: I will reuse connection in loadTablePrimaryKeys method

**Comment by zhujunxxxxx** [comment]
@tristaZero I solved the continuous integration testing problem by reusing the connection. 
@JiekerTime thanks for you advice, it's work.

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoaderTest.java [hunk 14]
```diff
@@ -0,0 +1,162 @@
...
+        return result;
+    }
+
+    private DialectTableMetaDataLoader getTableMetaDataLoader() {
+        for (DialectTableMetaDataLoader each : ShardingSphereServiceLoader.newServiceInstances(DialectTableMetaDataLoader.class)) {
+            if ("H2".equals(each.getDatabaseType())) {
+                return each;
+            }
+        }
+        throw new IllegalStateException("Can not find MySQLTableMetaDataLoader");
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoaderTest.java:151]: Is it `MySQLTableMetaDataLoader`?

**Review by tristaZero — CHANGES_REQUESTED** [review]

**Comment by tristaZero** [comment]
Hi @zhujunxxxxx ,

> I modify it.databases=MySQL in env/engine-env.properties and run GeneralDQLIT appear same problem. so, I am sure this it not my problem. It's just that the current scenario was not tested by the previous use case

You got this result from MySQL test case, didn't you? But @JiekerTime said, the `GeneralDQLIT` ran well in his env.
At least, CI tests passed. Still please look at my latest comment, thanks.

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoaderTest.java (hunk 14 above)
Comment (zhujunxxxxx) [reply shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/H2TableMetaDataLoaderTest.java:151]: clerical error，I will fix later.

**Comment by tristaZero** [comment]
@zhujunxxxxx Here is a Conflicting file.

**Comment by zhujunxxxxx** [comment]
@tristaZero Could you merge this pull reqeust,  I don't want to resolve conflicts anymore. ^_^!

**Review by tristaZero — APPROVED** [review]
Sure, thanks for your hard work!

**Pull request merged**